### PR TITLE
Integration of REST/API in client #4

### PR DESCRIPTION
**Pull request by bsekachev**
- Many client fixes:
    * Labels with white spaces and commas
    * Defiant
    * Splitting
    * Grouping 
- Repaired qunit tests for client
- Many codacy issues have been resolved

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -27,6 +27,14 @@
         "airbnb",
     ],
     "rules": {
+        "no-new": [0],
```
Comment (nmanovic) [inline .eslintrc.js]: Why do we need to disable rules? Why don't fix the code?

File: .eslintrc.js [hunk 2]
```diff
@@ -27,6 +27,14 @@
...
+        "class-methods-use-this": [0],
```
Comment (nmanovic) [inline .eslintrc.js]: Probably it is necessary to make methods as static if they don't use this?

File: .eslintrc.js [hunk 3]
```diff
@@ -27,6 +27,14 @@
...
+        "no-restricted-properties": [0, {
```
Comment (nmanovic) [inline .eslintrc.js]: I believe I already merged such changes? Do you need to merge develop into your branch?

**Comment by nmanovic** [comment]
@bsekachev , in the past we decided to use "full" version of js libraries because it is easy to debug problems. Now you are adding "min" versions. What is the reason for that? Does our "minimizer" work bad?

File: .eslintrc.js (hunk 1 above)
Comment (bsekachev) [reply .eslintrc.js]: @nmanovic 
We can't fix the code. If we assign an expression 'new Class()' to any variable which is unusable, we got eslint error "unusable variable" . If we don't assign this expression to a variable, we got eslint error "new is not needed". If we remove new, we got syntax error and code doesn't work.

File: .eslintrc.js (hunk 2 above)
Comment (bsekachev) [reply .eslintrc.js]: I thought about it. But there are many methods which is don't use "this" and if we do it static, they will become public available. I think it is not better way.

File: .eslintrc.js (hunk 3 above)
Comment (bsekachev) [reply .eslintrc.js]: I needed to do it in order to write unit tests. Probably small merge conflict will become here later, but I suppose it isn't big problem, to resolve it.

**Comment by bsekachev** [comment]
@nmanovic 
We don't have any minimizer yet, a bundler only. Now I don't see any problems in 3rd party libraries. If bugs occur, than they occur in our code. I believe to use minimized versions of js in production is right way and I am striving to it.

**Comment by nmanovic** [comment]
@bsekachev , if we don't have minimiser yet than it is OK. The right way is to use non-minimised version of all libraries and minimise them together with our code. Using minimised libraries is good only if you have a static page and don't have any pipeline to preprocess js. We have the pipeline but in the past we disabled minimiser because it worked incorrectly.

**Comment by bsekachev** [comment]
@nmanovic 
Okay, then we will use non-minimized versions in future.

**Pull request merged**